lta/rest_server.py

```diff
@@ -269,79 +276,130 @@ class FilesActionsPopHandler(BaseLTAHandler):
         dest = self.get_argument('dest')
         if dest not in self.sites:
             raise tornado.web.HTTPError(400, reason="invalid dest site")
+        # change the limit on the number of files that can be bundled
+        limit = self.get_argument('limit', 20000)
+        try:
+            limit = int(limit)
+        except Exception:
+            raise tornado.web.HTTPError(400, reason="limit is not an int")
+        # skip a number of files in the sorted results
+        skip = self.get_argument('skip', 0)
+        try:
+            skip = int(skip)
+        except Exception:
+            raise tornado.web.HTTPError(400, reason="skip is not an int")
         # figure out how big the bundle should be
-        bundle_size = self.sites[dest]["bundle_size"]
+        bundle_max_size = self.sites[dest]["bundle_size"]
         # get the self-identification of the claiming bundler
         pop_body = json_decode(self.request.body)
-        # let's build up a list of things to bundle
-        src_tuples = []
-        for uuid in self.db["Files"]:
-            db_file = self.db["Files"][uuid]
-            if not (site(db_file["source"]) == src):
-                continue
-            if not (site(db_file["dest"]) == dest):
-                continue
-            if not (db_file["status"] == "waiting"):
+        # let's find some files we can bundle
+        sdf = self.db.Files
+        query = {
+            "source": {"$regex": f"^{src}"},
+            "dest": {"$regex": f"^{dest}"},
+            "status": "waiting"
+        }
+        sort = [('catalog.logical_name', ASCENDING)]
+        bundle_files = []
+        bundle_full = False
+        bundle_size = 0
+        async for row in sdf.find(filter=query,
+                                  projection=REMOVE_ID,
+                                  skip=skip,
+                                  limit=limit,
```

Comment:
Don't limit the query, instead limit on the actual length of the bundle (maybe in line 324).